src/coreComponents/mesh/CellElementRegionSelector.cpp

```diff
@@ -0,0 +1,265 @@
+  }
+}
+
+
+void
+CellElementRegionSelector::registerRegionSelection( CellElementRegion const & region,
+                                                    std::set< integer > const & attributeValues,
+                                                    std::set< string > const & cellBlockNames )
+{
+  for( integer attributeValue : attributeValues )
+  {
+    string attributeValueStr = std::to_string( attributeValue );
+    m_regionAttributeOwners[attributeValueStr].push_back( &region );
+  }
+
+  for( string const & cellBlockName : cellBlockNames )
+  {
+    m_cellBlocksOwners[cellBlockName].push_back( &region );
+  }
+}
+
+
+std::set< string >
+CellElementRegionSelector::buildRegionCellBlocksSelection( CellElementRegion const & region )
+{
+  auto toStdSet = []( auto const & array ) {
+    using ArrayElementType = std::decay_t< decltype( *array.data() ) >;
+    return std::set< ArrayElementType >( array.begin(), array.end());
+  };
+
+  std::set< integer > const requestedAttributeValues = toStdSet(
+    region.getReference< integer_array >( ViewKeys::cellBlockAttributeValuesString() ) );
+  std::set< string > const requestedMatchPatterns = toStdSet(
+    region.getReference< string_array >( ViewKeys::cellBlockMatchPatternsString() ) );
+  std::set< string > const requestedCellBlock = toStdSet(
+    region.getReference< string_array >( ViewKeys::sourceCellBlockNamesString() ) );
+
+  // at least one selection method is needed
+  GEOS_THROW_IF( requestedAttributeValues.empty() && requestedMatchPatterns.empty() && requestedCellBlock.empty(),
+                 GEOS_FMT( "{}: cellBlocks must be selected to fill this region (using {}, {} or {}).",
+                           region.getDataContext(),
+                           ViewKeys::sourceCellBlockNamesString(),
+                           ViewKeys::cellBlockAttributeValuesString(),
+                           ViewKeys::cellBlockMatchPatternsString() ),
+                 InputError );
+  // only one selection method allowed for now
+  GEOS_THROW_IF( ( !requestedAttributeValues.empty() + !requestedMatchPatterns.empty() + !requestedCellBlock.empty() ) != 1,
+                 GEOS_FMT( "{}: Only one setting must be used to select cellBlocks ({}, {} or {}).",
+                           region.getDataContext(),
+                           ViewKeys::sourceCellBlockNamesString(),
+                           ViewKeys::cellBlockAttributeValuesString(),
+                           ViewKeys::cellBlockMatchPatternsString() ),
+                 InputError );
+
+  std::set< string > const matchPatterns = buildMatchPatterns( region, requestedAttributeValues, requestedMatchPatterns );
+  std::set< string > const matchedCellBlocks = getMatchingCellblocks( region, matchPatterns );
+  std::set< string > cellBlocksSelection = requestedCellBlock;
+  cellBlocksSelection.insert( matchedCellBlocks.begin(), matchedCellBlocks.end() );
+  verifyRequestedCellBlocks( region, cellBlocksSelection );
```

Comment:
Can you have both `requestedCellBlock` and `matchedCellBlocks` non empty (regarding the exclusive unique selection method you want)?
I would say that one has to be empty?

Reply:
Indeed, with the constraints on the multiple inputs, only one would be filled.
- This code would not change if the inputs constraints would change,
- Plus, I did this because, as I understood it, I had to avoid making branches, and the information would have to cascade through the same "pipes".

Reply:
Can you add a comment about it and why then?

Reply:
Done!